frappe/website/doctype/image_resize_preset/image_resize_preset.js

```diff
@@ -2,7 +2,19 @@
 // For license information, please see license.txt
 
 frappe.ui.form.on('Image Resize Preset', {
-	// refresh: function(frm) {
-
-	// }
+	refresh: function(frm) {
+		// view document button
+		if (!frm.is_new() && !frm.is_dirty()) {
+			frm.add_custom_button("Clear Cache", function() {
+				frappe.call({
+					method: "frappe.website.doctype.image_resize_preset.image_resize_preset.clear_cache",
+					args: { name: frm.doc.name },
+					freeze: true,
+					callback: function() {
+						frappe.msgprint(`Cache Cleared for ${frm.doc.name} Preset`)
+					}
+				})
```

Comment:
This is mostly a cleanup suggestion, but you could use the `frm.call` instead:

```suggestion
				frm.call({
					method: "clear_cache",
					doc: frm.doc,
					freeze: true,
					callback: function() {
						frappe.msgprint(`Cache Cleared for ${frm.doc.name} Preset`)
					}
				})
```

And on the server side, you can just move the function inside the class without the whitelist decorator

Reply:
Awesome, updated